From: Takashi Iwai <tiwai@suse.de>
Date: Mon, 26 Nov 2007 07:45:43 +0000 (+0100)
Subject: [ALSA] emu10k1x - Add missing snd_card_set_dev call
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fperex%2Falsa.git;a=commitdiff_plain;h=27ab27100d74f0f5073fc8d50dea6aefffe20dda

[ALSA] emu10k1x - Add missing snd_card_set_dev call

Added the missing snd_card_set_dev() call.  This will fix the incomplete
sysfs entry for this card.

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---

diff --git a/sound/pci/emu10k1/emu10k1x.c b/sound/pci/emu10k1/emu10k1x.c
index 1ec7eba..0030d8b 100644
--- a/sound/pci/emu10k1/emu10k1x.c
+++ b/sound/pci/emu10k1/emu10k1x.c
@@ -1583,6 +1583,8 @@ static int __devinit snd_emu10k1x_probe(struct pci_dev *pci,
 	sprintf(card->longname, "%s at 0x%lx irq %i",
 		card->shortname, chip->port, chip->irq);
 
+	snd_card_set_dev(card, &pci->dev);
+
 	if ((err = snd_card_register(card)) < 0) {
 		snd_card_free(card);
 		return err;
From: Takashi Iwai <tiwai@suse.de>
Date: Mon, 26 Nov 2007 07:58:48 +0000 (+0100)
Subject: [ALSA] drivers - Add missing snd_card_set_dev()
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fperex%2Falsa.git;a=commitdiff_plain;h=8e43d0d0c58a8705cb7c8fb5b5ec03efef8643f7

[ALSA] drivers - Add missing snd_card_set_dev()

Added the missing call of snd_card_set_dev() in drivers/*

Signed-off-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---

diff --git a/sound/drivers/mts64.c b/sound/drivers/mts64.c
index 68070cc..e12ba3d 100644
--- a/sound/drivers/mts64.c
+++ b/sound/drivers/mts64.c
@@ -1008,6 +1008,8 @@ static int __devinit snd_mts64_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, card);
 
+	snd_card_set_dev(card, &pdev->dev);
+
 	/* At this point card will be usable */
 	if ((err = snd_card_register(card)) < 0) {
 		snd_printd("Cannot register card\n");
diff --git a/sound/drivers/portman2x4.c b/sound/drivers/portman2x4.c
index 1b83287..7e7c668 100644
--- a/sound/drivers/portman2x4.c
+++ b/sound/drivers/portman2x4.c
@@ -797,6 +797,8 @@ static int __devinit snd_portman_probe(struct platform_device *pdev)
 
 	platform_set_drvdata(pdev, card);
 
+	snd_card_set_dev(card, &pdev->dev);
+
 	/* At this point card will be usable */
 	if ((err = snd_card_register(card)) < 0) {
 		snd_printd("Cannot register card\n");
